### PR TITLE
add presubmit job for dualstack

### PR DESCRIPTION
**Pull request by aramase**
- Adds presubmit job for dualstack tests on azure

cc @khenidak 

/assign @MrHohn

### CODE REVIEW

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:246]: I think that this is escaping a variable, the other jobs in this repo use `$AZURE_SSH_PUBLIC_KEY_FILE`, the other format should be with braces `${AZURE_SSH_PUBLIC_KEY_FILE}`, using parenthesis is to execute what is inside

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:240]: same comment about bash variables

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:234]: same comment

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: new tests have this tag `[Feature:IPv6DualStackAlphaFeature:Phase2]` , seems we have to use it without the trailing square bracket :shrug: 
```suggestion
        - --test_args=--ginkgo.focus=\[Feature:IPv6DualStackAlphaFeature
```

Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: actually, we should run more tests to be sure that all features works, this is very happy scenario

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:251]:
```suggestion
        - --ginkgo-parallel=20
```

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: once we verify this works, can you do the same changes for the periodic in a follow-up PR?

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:246]: @aojea All other azure jobs use `$(AZURE_SSH_PUBLIC_KEY_FILE)`. Where are you seeing `$AZURE_SSH_PUBLIC_KEY_FILE`?

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:221]: I think that this is for kind or others that use docker in docker  , let's ask @BenTheElder

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: >new tests have this tag [Feature:IPv6DualStackAlphaFeature:Phase2] , seems we have to use it without the trailing square bracket 🤷

So phase2 tests run too with `[Feature:IPv6DualStackAlphaFeature\]`- https://testgrid.k8s.io/provider-azure-dualstack#dualstack-azure-e2e-master.

Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: Yeah, +1 for running conformance. The periodic jobs initial focus was to just validate the dualstack e2e tests run as expected. We can start running it for presubmit and then port back the change to periodic.

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:255]: I never remember if you have to add this to the  `config/testgrids/config.yaml` 
Please, check that it has address to notify errors, it is mandatory or that was the intention to avoid stale jobs

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:246]: lol, I do not how old was the branch I was using but I can swear you the did not have it :smile: 
you are right, the rest of the repo has them this ways
TIL I honestly thought that same bash escaping rules applied here, it seems you can use parenthesis

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:240]: my bad

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:234]: my bad

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: yeah, but checks Khals PR 
https://github.com/kubernetes/kubernetes/pull/91824/commits/3d7ebf0fd96d2f5b78e77aba6bde2f4f074a2ecd#diff-62557e1c28188e811e7529d548cacb8cR352

we should change it here or there, can you both discuss it an agree on a solution?

Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: @aojea So this regexp matches `Feature:IPv6DualStackAlphaFeature` and `Feature:IPv6DualStackAlphaFeature:Phase2`. I tried that out locally.

Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: that is really weird, isn't it?

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:255]: Added 

```yaml
      testgrid-alert-email: kubernetes-provider-azure@googlegroups.com
```

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: image

Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: yeah, this is weird. 

```
➜ _output/local/bin/linux/amd64/e2e.test -ginkgo.focus="Feature:IPv6DualStackAlphaFeature" -ginkgo.dryRun
Ran 9 of 5228 Specs in 0.027 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 5219 Skipped
PASS
```
Also see  https://testgrid.k8s.io/provider-azure-dualstack#dualstack-azure-e2e-master. This runs few existing phase2 tests too with the same regex.

**Comment by aojea** [comment]
/lgtm
defer to @MrHohn for final approval, _I always miss something on prow jobs configs_

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: your regex does not have the square brackets, you have to add those 
--ginkgo.focus=\[Feature:IPv6DualStackAlphaFeature\] 

this way I think it will not match

Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: @aojea But that's the same regex used in the periodic job. That's running phase2 tests. 

```
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should be able to reach pod on ipv4 and ipv6 ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create pod, add ipv6 and ipv4 ip to pod ips
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with cluster ip from primary service range [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv4 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv6 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should have ipv4 and ipv6 internal node ip
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should have ipv4 and ipv6 node podCIDRs
```

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [inline config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: if we don't want to match on the square brackets is ok too, but is better to have them, also there is a typo because you are escaping the C in conformance
Actually, I would match in everything Feature dual stack, so we don't have to keep modifying this job
```suggestion
        - --test_args=--ginkgo.focus=\[Feature:IPv6DualStack.*\]|\[Conformance\]
```

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:221]: yep, I'm not sure what this job uses though.

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: the \[ escape the square bracket and match on them

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:221]: same for the security context below. if docker is not in use, it shouldn't be included. if it is, this is expected

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: duh, sorry @aramase this means I have to stop for today, 
```
[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv6 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]
```
it matches in the first tag

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: disregard this, the tests have two tags, but all of them have this one

**Comment by aojea** [comment]
/hold cancel
:bow:

File: config/jobs/kubernetes/sig-network/dualstack-e2e.yaml
Comment (aojea) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml:250]: @aramase the conformance tag is wrong though, it should be `\[Conformance\]` instead of `\[\Conformance\]`

Comment (aramase) [reply config/jobs/kubernetes/sig-network/dualstack-e2e.yaml]: @aojea Updated!

**Review by MrHohn — APPROVED** [review]
> defer to @MrHohn for final approval, I always miss something on prow jobs configs

Well I'm definitely not an expert on the prow configs, but this looks sane to me :)

/approve

**Pull request merged**
